src/main/kotlin/me/zeroeightsix/kami/feature/module/AutoEat.kt

```diff
@@ -1,21 +1,44 @@
 package me.zeroeightsix.kami.feature.module
 
+import io.github.fablabsmc.fablabs.api.fiber.v1.annotation.Setting
 import me.zero.alpine.listener.EventHandler
 import me.zero.alpine.listener.Listener
 import me.zeroeightsix.kami.event.TickEvent.InGame
 import me.zeroeightsix.kami.mixin.client.IKeyBinding
+import me.zeroeightsix.kami.setting.SettingVisibility
 import net.minecraft.client.options.KeyBinding
 import net.minecraft.item.ItemGroup
 import net.minecraft.item.ItemStack
 import net.minecraft.util.Hand
+import kotlin.math.roundToInt
 
 @Module.Info(name = "AutoEat", description = "Automatically eat when hungry", category = Module.Category.PLAYER)
 object AutoEat : Module() {
     private var eating: Hand? = null
     private var oldSlot: Int? = null
 
+    @Setting(comment = "Use the threshold value to activate AutoEat instead of the food hunger value")
+    private var useCustomThreshold = false
+
+    @Setting(comment = "If the amount of free hunger points goes above this, AutoEat activates")
+    @SettingVisibility.Method("shouldUseCustomThreshold")
+    // there is absolutely no reason for this to be a double but i can't use an Int
+    private var threshold: @Setting.Constrain.Range(min = 0.0, max = 20.0, step = 1.0) Double = 5.0
```

Comment:
`Int` is a valid setting type as well. Why could you not use it?

Reply:
i got an error but now that im looking at it it may also be that i initially was missing step